mariamihaelafainarea/assignment2/LowestAncestor

```diff
@@ -50,20 +48,20 @@ public class LowestAncestor<T> {
             this.right = right;
         }
     }
-    private Node findLowestAncestor(T x, Node root,T y){
+    private Node findLowestAncestor(T x, Node root,T y,Boolean[] params){
```

Comment:
Okay that's already better :) But maybe try to pass two booleans instead of an array of Booleans? The reason for it is that with an array it can be easily misused: someone can pass a longer or shorter array and you still will not have the params that you expect. It is always supposed to be two boolean values, in every run of the findLowestAncestor method, so just add them as the expected parameters.

I agree with you that looking for the values beforehand will be slower. But it would still be better than introducing a possible bug :)

Reply:
Okey. I have solved it by introducing two AtomicBooleans because you cannot modify the value of a boolean by sending it to another function. if i have param1 = true;param2 = true; and i call the function change(param1,param2) { param1 = false; param2 = false;} after the end of the function the values will be param1 = true; param2 = true; That was the reason that i used an array of booleans. The only method is two introduce AtomicBoolean as it is written here :
https://stackoverflow.com/questions/24271949/how-we-can-get-boolean-value-updated-through-other-method-in-main-method